### PR TITLE
fix(recipe): fix deprecation warning from threading.Event

### PR DESCRIPTION
**Pull request by stefreak**
Fix deprecation warning emitted when calling `threading.Event.isSet`

## Why is this needed?

Reasoning:
- `is_set` is supported since Python2.6 (https://docs.python.org/2.7/library/threading.html)
- Starting from Python3.5 docs the `isSet` spelling is not even mentioned anymore (https://docs.python.org/3.5/library/threading.html)
- Python3.10 will give a deprecation warning when the `isSet` spelling is used (https://docs.python.org/3.10/library/threading.html)

## Proposed Changes

- just use is_set over isSet

## Does this PR introduce any breaking change?

- only for users of python version below 2.6; python2.5 is not supported according to setup.py

### CODE REVIEW

**Comment by stefreak** [comment]
I'll be glad to rebase once #652 is merged. Just let me know if I overlook it :)

**Review by ceache — CHANGES_REQUESTED** [review]
Hi, thank you for your contribution. It looks good to me.  I would only ask that you amend your PR to follow our contribution guideline (https://github.com/python-zk/kazoo/blob/master/CONTRIBUTING.md#git-commit-guidelines) and I would be happy to approve it.

**Comment by stefreak** [comment]
@ceache sure, thanks a lot for the feedback! I have changed the commit message accordingly.

**Review by StephenSorriaux — APPROVED** [review]
Thank you!

**Review by ceache — APPROVED** [review]

**Comment by stefreak** [comment]
I rebased with master, hoping that it would run the github actions for my PR, but it didn't; sorry for dismissing your reviews @ceache and @StephenSorriaux

**Comment by StephenSorriaux** [comment]
Sorry about that, that is actually on me. I will fix it.

**Review by ceache — DISMISSED** [review]

**Comment by StephenSorriaux** [comment]
@stefreak Can you please rebase your PR with the current master that fixes the CI part? Thank you.

**Comment by stefreak** [comment]
@StephenSorriaux done :)

**Review by StephenSorriaux — APPROVED** [review]

**Review by ceache — APPROVED** [review]

**Pull request merged**
